js/app.js

```diff
@@ -52,6 +52,93 @@ var Textile = {
     },
     parseZip: function(zip) {
         console.log("parsing: zip data");
+        if (zip.file("html/photos.htm")) {
+            Textile.parseHtml(zip);
+        }
+        else {
+            Textile.parseJson(zip);
+        }
+    },
+    parseJsonYourPosts: function(mapper) {
+        var status_updates = mapper["status_updates"];
+        status_updates.forEach(function (status_update) {
+            var milliUnix = 1000 * status_update["timestamp"];
+            var attachments = status_update["attachments"];
+            var isoDate = new Date(milliUnix).toISOString();
+            if (attachments) {
+                var statusFileCount = attachments.length;
+                var msg = " files in a status on ";
+                console.log(statusFileCount + msg + isoDate);
+            }
+            return;
+        });
+        return true
+    },
+    parseZipMap: function(zip, mapper) {
+        zip.forEach(function (relativePath, zipEntry) {
+            var name;
+            if (relativePath.match(/^photos\/\S*\/\S*jpg/)) {
+                var index = relativePath.lastIndexOf("/");
+                var base = mapper[relativePath.substring(0, index)];
+                var photo = relativePath.substring(index);
+                if (base === undefined) {
+                    base = relativePath.substring(0, index)
+                }
+                Textile.file.file("Photos/" + base + photo, zipEntry._data)
+            }
+
+            if (relativePath.match(/^messages\/\S*jpg/)) {
+                var splitPath = relativePath.split("/");
+                name = splitPath[splitPath.length - 1];
+                Textile.file.file("Photos/Messages/"+name, zipEntry._data)
+            }
+        });
+        return true
+    },
```

Comment:
Add reusable function called Textile.parseZipMap
⌛️Use Textile.parseZipMap in Textile.parseJson
⏳Reuse Textile.parseZipMap in Textile.parseHtml

Reply:
So we "Don't repeat ourselves", we could use the new `parseZipMap` function in `parseHtml`, but that may call for a separate pull request with separate testing.

You can accept this PR without a decision on this.